cpp/src/arrow/compute/exec/sink_node.cc

```diff
@@ -188,7 +193,12 @@ class ConsumingSinkNode : public ExecNode {
       return;
     }
 
-    Status consumption_status = consumer_->Consume(std::move(batch));
+    auto batch = task();
+    if (!batch.ok()) {
+      ErrorIfNotOk(batch.status());
+      return;
+    }
+    Status consumption_status = consumer_->Consume(batch.MoveValueUnsafe());
```

Comment:
This changes to submit task.

Reply:
SubmitTask is a method that is going to be part of the interface of the Scheduler, this part will need to changes in the next PRs related to the scheduler x2